crates/subspace-archiving/src/archiver.rs

```diff
@@ -202,18 +202,33 @@ pub struct ArchiveBlockOutcome {
 #[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, thiserror::Error)]
 pub enum ArchiverInstantiationError {
     /// Invalid last archived block, its size is the same as encoded block
-    #[error("Invalid last archived block, its size {0} bytes is the same as encoded block")]
-    InvalidLastArchivedBlock(BlockNumber),
+    #[error(
+        "Invalid last archived block, its size {block_bytes} bytes is the same as encoded \
+        block, archived in segment: {prev_segment_index:?} {prev_segment_header_hash:?}"
+    )]
+    InvalidLastArchivedBlock {
+        /// Already archived block size, which is equal to the full block size
+        block_bytes: u32,
+        /// The segment index for the segment that already archived the block
+        prev_segment_index: SegmentIndex,
+        /// The segment header hash for the segment that already archived the block
+        prev_segment_header_hash: Blake3Hash,
+    },
```

Comment:
Not sure how useful this actually is, we already have this information on higher level.

I'd probably replace "Incorrect parameters for archiver" panic with an error message that also logs the whole `last_segment_header`, which will have the same effect.